medicines/doc-index-updater/src/service_bus_client/mod.rs

```diff
@@ -63,15 +64,33 @@ impl Error for RetrieveFromQueueError {
     }
 }
 
+pub struct RetrievedMessage<T> {
+    pub message: T,
+    peek_lock: PeekLockResponse,
+}
+
+impl<T> RetrievedMessage<T> {
+    pub async fn remove(&self) -> Result<String, anyhow::Error> {
+        let queue_removal_result = self.peek_lock.delete_message().await.map_err(|e| {
+            tracing::error!("{:?}", e);
+            anyhow!("Queue Removal Error")
+        });
```

Comment:
It's way out of scope for this ticket, but it's annoying we have to do this all over the place.

I don't suppose there's a way we could implement this as a trait on the `AzureError` somewhere?

Reply:
We can't because we don't own `AzureError` or std::error::Error`. You can only `impl` things that do both. We could wrap it like `MyRedisError` does, and implement for that. What I do like about doing this `anyhow!` here is that I can see the error message.